jelares/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,99 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.ExampleComments;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that manages comments page */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  // Make the comments list data type
+  final ExampleComments comments = new ExampleComments(new ArrayList<String>());
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jesus!</h1>");
+    // Read the query string to get comment limit
+    int commentLimit = Integer.parseInt(request.getParameter("num-comments"));
+
+    // Query to find all comment entities sorted from newest to oldest
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // adding all comment entities to an ExampleComments instance
+    final ExampleComments comments = new ExampleComments(new ArrayList<String>());
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(commentLimit))) {
```

Comment:
Do you know what's the difference between setting the limit here vs setting it on the Query?
https://cloud.google.com/datastore/docs/concepts/queries#datastore-datastore-limit-java

I'm wondering which is more applicable to this situation.

Reply:
AnneMarie pointed out that that API is unavailable to your environment. It does look like `withLimit()` is most applicable. Glad you found this!

Reply:
I have been trying to fix this for a few hours but cannot find any way to set the limit on Query itself. The link you sent me ended up being for the cloud datastore version, but for our project we are using the appengine datastore version - the documentation for it is here: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query.html

However I have looked through it and cannot find any ways to set limit on Query objects. I'll keep looking tomorrow, though.